sema4ai/vscode-client/src/robo/agentPackage.ts

```diff
@@ -73,3 +77,80 @@ export const createAgentPackage = async (): Promise<void> => {
         window.showErrorMessage(errorMsg + " (see `OUTPUT > Sema4.ai` for more details).");
     }
 };
+
+export const packAgentPackage = async (): Promise<void> => {
+    const agentCli = getAgentCliLocation();
+    if (!agentCli) {
+        return;
+    }
+
+    let ws: WorkspaceFolder | undefined = await askForWs();
+    if (!ws) {
+        // Operation cancelled.
+        return;
+    }
+
+    const [rootPackageYaml, workspacePackages] = await Promise.all([
+        getPackageYamlNameFromDirectory(ws.uri),
+        getWorkspacePackages(),
+    ]);
+
+    if (!workspacePackages?.agentPackages?.length) {
+        window.showErrorMessage("No Agent Packages found in the workspace.");
+        return;
+    }
+
+    let packageInfo: LocalPackageMetadataInfo;
+    let actionPackageSelection: string | null = null;
+
+    /* If root level contains an agent-spec.yaml, there will be only one Agent Package. */
+    if (rootPackageYaml === PackageYamlName.Agent) {
+        packageInfo = workspacePackages.agentPackages[0];
+    } else {
+        actionPackageSelection = await window.showQuickPick(
+            workspacePackages.agentPackages.map((agentPackage) => agentPackage.name),
+            {
+                placeHolder: "Which action package do you want to pack?",
+                ignoreFocusOut: true,
+            }
+        );
+
+        /* Operation cancelled. */
+        if (!actionPackageSelection) {
+            return null;
+        }
+
+        // Update packageInfo based on selected action package
+        packageInfo = workspacePackages.agentPackages.find(
+            (agentPackage) => agentPackage.name === actionPackageSelection
+        );
+    }
+
+    const targetDir = packageInfo.directory;
+
+    try {
+        const result: ActionResult<string> = await commands.executeCommand(
+            roboCommands.SEMA4AI_PACK_AGENT_PACKAGE_INTERNAL,
+            { directory: targetDir }
+        );
+
+        if (!result.success) {
+            throw new Error(result.message || "An unexpected error occurred during the packaging process.");
+        }
+
+        refreshFilesExplorer();
+        window.showInformationMessage(`Package successfully packed at:\n${targetDir}`);
+    } catch (error) {
+        const errorMsg = `Failed to package the agent at: ${targetDir}`;
+        logError(errorMsg, error, "ERR_PACK_ACTION_PACKAGE");
+
+        const detailedErrorMsg = `${errorMsg}. Please check the 'OUTPUT > Sema4.ai' for more details.`;
+        OUTPUT_CHANNEL.appendLine(detailedErrorMsg);
+        window.showErrorMessage(detailedErrorMsg);
+    }
+
+    const zipUri = Uri.file(path.join(targetDir, "agent-package.zip"));
```

Comment:
Maybe the `ActionResult<string>` could actually have the path to the generated file and then here you wouldn't hardcode the `agent-package.zip` in the client side?

Reply:
That's a good ideea indeed